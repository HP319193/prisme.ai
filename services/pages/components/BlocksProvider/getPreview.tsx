import {
  AudioOutlined,
  FilePdfOutlined,
  FileTextOutlined,
  FileUnknownOutlined,
  FileWordOutlined,
  FileExcelOutlined,
  FilePptOutlined,
} from '@ant-design/icons';

export const getPreview = (mimetype: string, url: string) => {
  const [type] = mimetype.split(/\//);
  if (type === 'image') {
    return url;
  }

  if (mimetype === 'application/pdf') {
    return (
      <FilePdfOutlined className="text-4xl !text-accent flex items-center" />
    );
  }
  if (
    mimetype.includes('officedocument.wordprocessingml') ||
    mimetype.includes('msword') ||
    mimetype.includes('ms-word')
  ) {
    return (
      <FileWordOutlined className="text-4xl !text-accent flex items-center" />
    );
  }

  if (
    mimetype.includes('officedocument.spreadsheetml') ||
    mimetype.includes('ms-excel')
  ) {
    return (
      <FileExcelOutlined className="text-4xl !text-accent flex items-center" />
    );
  }

  if (
    mimetype.includes('officedocument.presentationml') ||
    mimetype.includes('ms-powerpoint')
  ) {
    return (
      <FilePptOutlined className="text-4xl !text-accent flex items-center" />
    );
  }
  if (type === 'audio') {
    return (
      <AudioOutlined className="text-4xl !text-accent flex items-center" />
    );
  }
  if (type === 'text') {
    return (
      <FileTextOutlined className="text-4xl !text-accent flex items-center" />
    );
  }

  return (
    <FileUnknownOutlined className="text-4xl !text-accent flex items-center" />
  );
};
