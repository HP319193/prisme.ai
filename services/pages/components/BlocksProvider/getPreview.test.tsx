import { getPreview } from './getPreview';
import renderer from 'react-test-renderer';
import {
  FilePdfOutlined,
  FileWordOutlined,
  AudioOutlined,
  FileTextOutlined,
  FileUnknownOutlined,
} from '@ant-design/icons';
jest.mock('@ant-design/icons', () => ({
  FilePdfOutlined: function FilePdfOutlined() {
    return null;
  },
  FileWordOutlined: function FileWordOutlined() {
    return null;
  },
  AudioOutlined: function AudioOutlined() {
    return null;
  },
  FileTextOutlined: function FileTextOutlined() {
    return null;
  },
  FileUnknownOutlined: function FileUnknownOutlined() {
    return null;
  },
}));
it('should get image preview', () => {
  expect(getPreview('image/jpg', 'file.jpg')).toEqual('file.jpg');
});

it('should get pdf preview', () => {
  const preview = getPreview('application/pdf', 'file.pdf') as JSX.Element;
  expect(preview.type).toEqual(FilePdfOutlined);
  const root = renderer.create(preview);
  expect(root.toJSON()).toMatchSnapshot();
});

it('should get office preview', () => {
  const preview = getPreview(
    'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
    'file.doc'
  ) as JSX.Element;
  expect(preview.type).toEqual(FileWordOutlined);
  const root = renderer.create(preview);
  expect(root.toJSON()).toMatchSnapshot();
});

it('should get audio preview', () => {
  const preview = getPreview('audio/mp3', 'file.mp3') as JSX.Element;
  expect(preview.type).toEqual(AudioOutlined);
  const root = renderer.create(preview);
  expect(root.toJSON()).toMatchSnapshot();
});

it('should get text preview', () => {
  const preview = getPreview('text/plain', 'file.text') as JSX.Element;
  expect(preview.type).toEqual(FileTextOutlined);
  const root = renderer.create(preview);
  expect(root.toJSON()).toMatchSnapshot();
});

it('should get anything else preview', () => {
  const preview = getPreview(
    'application/octet-stream',
    'file.bin'
  ) as JSX.Element;
  expect(preview.type).toEqual(FileUnknownOutlined);
  const root = renderer.create(preview);
  expect(root.toJSON()).toMatchSnapshot();
});
