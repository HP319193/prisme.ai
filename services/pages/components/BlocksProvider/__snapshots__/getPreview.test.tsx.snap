// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get anything else preview 1`] = `null`;

exports[`should get audio preview 1`] = `null`;

exports[`should get office preview 1`] = `null`;

exports[`should get pdf preview 1`] = `null`;

exports[`should get text preview 1`] = `null`;
